src/main/java/org/graylog2/plugins/slack/callback/SlackAlarmCallback.java

```diff
@@ -133,9 +132,52 @@ public String buildMessage(Stream stream, AlertCondition.CheckResult result) {
             titleLink = "_" + stream.getTitle() + "_";
         }
 
-        return (notifyChannel ? "@channel " : "") + "*Alert for Graylog stream " + titleLink + "*:\n" + "> " + result.getResultDescription();
+        // Build custom message
+        StringBuilder message = new StringBuilder(result.getResultDescription()).append("\n");
+        String template = configuration.getString(SlackConfiguration.CK_CUSTOM_MESSAGE);
+        if (!isNullOrEmpty(template)) {
+            String customMessage = buildCustomMessage(stream, result, template);
+            message.append("\n").append(customMessage);
+        }
+
+        String audience = notifyChannel ? "@channel " : "";
+        return String.format("%s*Alert for Graylog stream %s*:\n> %s",
+                audience, titleLink, message.toString());
+    }
+
+    private String buildCustomMessage(Stream stream, AlertCondition.CheckResult result, String template) {
+        List<Message> backlog = getAlarmBacklog(result);
+        Map<String, Object> model = getModel(stream, result, backlog);
+        try {
+            return templateEngine.transform(template, model);
+        } catch (Exception ex) {
+            return ex.toString();
+        }
+    }
+
+    private Map<String, Object> getModel(Stream stream, AlertCondition.CheckResult result, List<Message> backlog) {
+        Map<String, Object> model = new HashMap<>();
+        String graylogUri = configuration.getString(SlackConfiguration.CK_GRAYLOG2_URL);
+        model.put("stream", stream);
+        model.put("check_result", result);
+        if (!isNullOrEmpty(graylogUri)) model.put("stream_url", buildStreamLink(graylogUri, stream));
+        model.put("alert_condition", result.getTriggeredCondition());
+        model.put("backlog", backlog);
+        model.put("backlog_size", backlog.size());
+
+        // Load all matched fields
+        result.getMatchingMessages().forEach(msg -> {
```

Comment:
This makes all matched message field/values available for templating. Using an indexer like syntax to get field/values when there is more than one message (multiple instances of the same field).